agent.go

```diff
@@ -188,11 +188,35 @@ func (c *container) deleteProcess(execID string) {
 	c.Unlock()
 }
 
+func (c *container) waitContainer() error {
+	for i := 0; i < 100; i++ {
+		if err := c.container.Signal(syscall.Signal(0), false); err != nil {
+			return nil
+		}
+		time.Sleep(100 * time.Millisecond)
+	}
+
+	return fmt.Errorf("container is still running")
+}
+
 func (c *container) removeContainer() error {
+	s, err := c.container.Status()
+	if err != nil {
+		return err
+	}
+
+	// Wait a bit to see if container stops. This might happen if
```

Comment:
well, 10 seconds is not a bit

Reply:
Referencing [runc](https://github.com/opencontainers/runc/blob/master/delete.go#L18) :)